File: 1. JPA basic/basic/src/main/java/study/jpa/basic/jpashop/domain/Book.java
package study.jpa.basic.jpashop.domain;

import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * @author Joonhyuck Hyoung
 */
@Entity(name = "jpashop_book")
@Table(name = "jpashop_book")
public class Book extends Item {
    private String author;
    private String isbn;

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }
}

File: 2. JPA deep/jpashop/src/main/kotlin/jpabook/jpashop/domain/item/Book.kt
package jpabook.jpashop.domain.item

import jpabook.jpashop.domain.*
import javax.persistence.DiscriminatorValue
import javax.persistence.Entity

/**
 * @author Joonhyuck Hyoung
 */
@Entity
@DiscriminatorValue("BOOK")
class Book(
    val author: String,
    val isbn: String,

    name: String,
    price: Int,
    stockQuantity: Int,
) : Item(name, price, stockQuantity)
